solo-chains/runtime/starlight/tests/collator_assignment_tests.rs

```diff
@@ -0,0 +1,615 @@
+                vec![alice_keys.babe.clone(), bob_keys.babe.clone()]
+            );
+
+            // Change Alice and Bob keys to something different
+            // for now lets change it to alice_keys_2 and bob_keys_2
+            assert_ok!(Session::set_keys(
+                origin_of(ALICE.into()),
+                starlight_runtime::SessionKeys {
+                    babe: alice_keys_2.babe.clone(),
+                    grandpa: alice_keys_2.grandpa.clone(),
+                    para_validator: alice_keys_2.para_validator.clone(),
+                    para_assignment: alice_keys_2.para_assignment.clone(),
+                    authority_discovery: alice_keys_2.authority_discovery.clone(),
+                    beefy: alice_keys_2.beefy.clone(),
+                    nimbus: alice_keys_2.nimbus.clone(),
+                },
+                vec![]
+            ));
+
+            assert_ok!(Session::set_keys(
+                origin_of(BOB.into()),
+                starlight_runtime::SessionKeys {
+                    babe: bob_keys_2.babe.clone(),
+                    grandpa: bob_keys_2.grandpa.clone(),
+                    para_validator: bob_keys_2.para_validator.clone(),
+                    para_assignment: bob_keys_2.para_assignment.clone(),
+                    authority_discovery: bob_keys_2.authority_discovery.clone(),
+                    beefy: bob_keys_2.beefy.clone(),
+                    nimbus: bob_keys_2.nimbus.clone(),
+                },
+                vec![]
+            ));
+
+            run_to_session(1u32);
+            let key_mapping_session_0 = TanssiAuthorityMapping::authority_id_mapping(0).unwrap();
+            assert_eq!(key_mapping_session_0.len(), 2);
+            assert_eq!(
+                key_mapping_session_0.get(&alice_keys.nimbus),
+                Some(&ALICE.into())
+            );
+            assert_eq!(
+                key_mapping_session_0.get(&bob_keys.nimbus),
+                Some(&BOB.into())
+            );
+
+            let key_mapping_session_1 = TanssiAuthorityMapping::authority_id_mapping(1).unwrap();
+
+            assert_eq!(key_mapping_session_1.len(), 2);
+            assert_eq!(
+                key_mapping_session_1.get(&alice_keys_2.nimbus),
+                Some(&ALICE.into())
+            );
+            assert_eq!(
+                key_mapping_session_1.get(&bob_keys_2.nimbus),
+                Some(&BOB.into())
+            );
+
+            // Let's check Babe again
+            assert_eq!(
+                authorities(),
+                vec![alice_keys.babe.clone(), bob_keys.babe.clone()]
+            );
+
+            run_to_session(2u32);
+            assert!(TanssiAuthorityMapping::authority_id_mapping(0).is_none());
+
+            let key_mapping_session_1 = TanssiAuthorityMapping::authority_id_mapping(1).unwrap();
+            assert_eq!(key_mapping_session_1.len(), 2);
+            assert_eq!(
+                key_mapping_session_1.get(&alice_keys_2.nimbus),
```

Comment:
Is this behavior expected? The test reflects that the new keys are placed inside the mapping for session 1, but shouldn't them be applied after session 2?

Reply:
solved